<launch>
    <arg name = "veh" doc = "duckiebot"/>
    <arg name = "pkg_name" value = "motion_planner"/>
    <arg name = "node_name" default = "extended_kalman_filter_node"/>
    <arg name = "parameter_file_name" default = "default"/>
    <group ns = "$(arg veh)">
        <remap from = "extended_kalman_filter_node/initialise" to = "matlab/initialise"/>
        <remap from = "extended_kalman_filter_node/apriltag_measurement" to = "apriltag_detector_node/apriltag_measurement"/>
        <remap from = "extended_kalman_filter_node/prediction" to = "odometry_node/prediction"/>
        <node name = "$(arg node_name)" pkg = "$(arg pkg_name)" type = "$(arg node_name)" output = "screen">
            <param name = "veh" value = "$(arg veh)"/>
            <rosparam command = "load" file = "$(find motion_planner)/config/$(arg node_name)/$(arg parameter_file_name).yaml"/>
        </node>
    </group>
</launch>
